transit_odp/fares_validator/schema/fares_schema.json

```diff
@@ -280,280 +280,114 @@
+      "context": "//x:dataObjects",
       "rules": [
         {
-          "test": "count(x:distributionAssignments) > 0"
+          "test": "is_uk_pi_fare_price_frame_present(.)"
         }
       ]
     },
     {
-      "details": "Mandatory element 'DistributionAssignment' missing",
+      "details": "Mandatory element 'fareProducts' missing for FareFrame - UK_PI_FARE_PRODUCT",
       "category": "Mandatory Element",
-      "context": "//x:salesOfferPackages/x:SalesOfferPackage/x:distributionAssignments",
+      "context": "//x:dataObjects",
       "rules": [
         {
-          "test": "count(x:DistributionAssignment) > 0"
+          "test": "check_fare_products(.)"
         }
       ]
     },
     {
-      "details": "Mandatory element 'DistributionChannelType' missing",
+      "details": "Mandatory element 'PreassignedFareProduct' or it's children missing in fareProducts for FareFrame - UK_PI_FARE_PRODUCT",
       "category": "Mandatory Element",
-      "context": "//x:salesOfferPackages/x:SalesOfferPackage/x:distributionAssignments/x:DistributionAssignment",
+      "context": "//x:dataObjects",
       "rules": [
         {
-          "test": "count(x:DistributionChannelType) > 0"
+          "test": "check_preassigned_fare_products(.)"
         }
       ]
     },
     {
-      "details": "Mandatory element 'PaymentMethods' missing",
+      "details": "Mandatory element 'validableElements' or it's children missing in fareProducts.PreassignedFareProduct for FareFrame - UK_PI_FARE_PRODUCT",
       "category": "Mandatory Element",
-      "context": "//x:salesOfferPackages/x:SalesOfferPackage/x:distributionAssignments/x:DistributionAssignment",
+      "context": "//x:dataObjects",
       "rules": [
         {
-          "test": "count(x:PaymentMethods) > 0"
+          "test": "check_preassigned_validable_elements(.)"
         }
       ]
     },
     {
-      "details": "Mandatory element 'salesOfferPackageElements' missing",
+      "details": "Mandatory element 'AccessRightInProduct' or it's children missing in fareProducts.PreassignedFareProduct for FareFrame - UK_PI_FARE_PRODUCT",
       "category": "Mandatory Element",
-      "context": "//x:salesOfferPackages/x:SalesOfferPackage",
+      "context": "//x:dataObjects",
       "rules": [
         {
-          "test": "count(x:salesOfferPackageElements) > 0"
+          "test": "check_access_right_elements(.)"
         }
       ]
     },
     {
-      "details": "Mandatory element 'SalesOfferPackageElement' missing",
+      "details": "Mandatory element 'ProductType' is missing in fareProducts.PreassignedFareProduct for FareFrame - UK_PI_FARE_PRODUCT",
       "category": "Mandatory Element",
-      "context": "//x:salesOfferPackages/x:SalesOfferPackage/x:salesOfferPackageElements",
+      "context": "//x:dataObjects",
       "rules": [
         {
-          "test": "count(x:SalesOfferPackageElement) > 0"
+          "test": "check_product_type(.)"
```

Comment:
This can be combined with check_preassigned_fare_products since it's 'PreassignedFareProduct's child